stellar-contract-env-host/src/host.rs

```diff
@@ -867,10 +870,54 @@ impl CheckedEnv for Host {
     }
 
     fn compute_hash_sha256(&self, x: Object) -> Result<Object, HostError> {
-        todo!()
+        let binary = self.visit_obj(x, |bin: &Vec<u8>| {
+            Ok(bin.clone()) //TODO:Why is this neccesary
+        })?;
+
+        let mut hasher = Sha256::new();
+        hasher.update(binary);
+        let raw_hash = hasher.finalize();
+
+        let hash: Vec<u8> = raw_hash[0..32]
+            .try_into()
+            .expect("slice with incorrect length");
+        Ok(self.add_host_object(hash)?.into())
     }
 
     fn verify_sig_ed25519(&self, x: Object, k: Object, s: Object) -> Result<RawVal, HostError> {
-        todo!()
+        let key_bytes = self.visit_obj(k, |bin: &Vec<u8>| {
+            let arr: [u8; 32] = bin
+                .as_slice()
+                .try_into()
+                .map_err(|_| HostError::General("invalid raw key"))?;
+            Ok(arr.clone())
```

Comment:
The .clone here is unnecessary.
```suggestion
            Ok(arr)
```

Reply:
removed